_posts/2020/2020-10-30-pmemkv-java-binding.md

```diff
@@ -0,0 +1,137 @@
+---
+title: API overview of pmemkv-java binding
+author: karczex
+layout: post
+identifier: pmemkv-java-binding
+---
+
+Pmemkv is a key-value data store written in C and C++, however, it also opens up
+a way to leverage persistent memory by developers who  prefer high-level languages - such as Java.
+For more information about other bindings please read [Language bindings for pmemkv][pmemkv-bindings]
+article and [pmemkv README][pmemkv-bindings-readme]
+
+We built an API for pmemkv-java bindings on top of libpmemkv 1.0 API but java-binding
+is also compatible with newer versions of libpmemkv. Hence, you don’t have to worry
+about the dependencies, because packages from the OS vendor’s repositories suffice.
+
+### Examples
+
+[Examples directory][examples] in pmemkv-java repository is a good resource, where you may learn how to
+create pmemkv database and use it with different types of data; starting from simple [ByteBuffer][ByteBufferExample],
+up to little bit more [real-life-application][PicturesExample], which shows you how to store and display pictures.
+
+#### Implementing Converter class
+
+Since pmemkv-java binding is the java wrapper for c++ library, it internally relies on java native interface,
+so it allows to store only ByteBuffer objects. It wouldn't be very convenient to operate on ByteBuffers inside
+a high level application, so we introduced an additional layer of types converters. It's possible to store instances of any class,
```

Comment:
`Because it wouldn't be ..., ... we've introduced ... type converters. It makes it possible to store an instance of any class in pmemkv, as long as an appropriate converter is available.`

(is "Type Converter" a java term? sounds odd)